### PR TITLE
Implement `use_compat()`

### PR DESCRIPTION
**Pull request by DavisVaughan**
Used like `use_compat("r-lib/rlang", "purrr")`

- Doesn't overwrite by default
- Requires `r-lib/rlang` to have a `main` branch. This is where we download the source code from.
- This is an internal function

### CODE REVIEW

**Comment by DavisVaughan** [comment]
Decide if https://usethis.r-lib.org/reference/use_github_file.html is good enough instead

**Comment by lionel-** [comment]
I think we'll still need a wrapper around `use_github_file()` so that we can build a simple dependency system into it. I'm thinking about something like this:

```r
.compat_deps <- c("compat-rlang.R", "compat-cli.R")
```

We'd import/update these deps as well recursively.

File: R/c-lib.R [hunk 1]
```diff
@@ -77,16 +77,24 @@ use_rlang_c_library <- function() {
   }
 }
 
-download_rlang <- function() {
-  dest_zip <- fs::file_temp("rlang-src")
+download_package <- function(owner, repo) {
+  # Hardcoded for now
+  branch <- "main"
```
Comment (jennybc) [inline R/c-lib.R:82]: In general, you can often skirt the `master` vs. `main` issue by using `HEAD` as the ref.

**Comment by lionel-** [comment]
Closed by https://github.com/r-lib/usethis/pull/1748

**Pull request closed without merging**